source/UberCoreTests/AuthenticationURLUtilityTests.swift

```diff
@@ -96,20 +95,13 @@ class AuthenticationURLUtilityTests: XCTestCase {
     }
     
     func testShouldHandleRedirectURL() {
-        let testRedirectURLString = "test://handleThis"
-        guard let testRedirectURL = URL(string: testRedirectURLString) else {
-            XCTFail()
-            return
-        }
-        Configuration.shared.setCallbackURIString(testRedirectURLString, type: .implicit)
-        XCTAssertFalse(AuthenticationURLUtility.shouldHandleRedirectURL(testRedirectURL, type: .general))
-        XCTAssertFalse(AuthenticationURLUtility.shouldHandleRedirectURL(testRedirectURL, type: .native))
-        XCTAssertFalse(AuthenticationURLUtility.shouldHandleRedirectURL(testRedirectURL, type: .authorizationCode))
+        let testRedirectURL = URL(string: "test://handleThis")!
         
-        XCTAssertTrue(AuthenticationURLUtility.shouldHandleRedirectURL(testRedirectURL, type: .implicit))
+        Configuration.shared.setCallbackURI(testRedirectURL, type: .general)
+        XCTAssertTrue(AuthenticationURLUtility.shouldHandleRedirectURL(testRedirectURL))
         
-        Configuration.shared.setCallbackURIString(nil, type: .implicit)
+        Configuration.shared.setCallbackURI(nil, type: .general)
```

Comment:
Not obvious to me what this change does and why it is bundled with the rest of this diff.

Reply:
This is related to the configuration changes in #203 where we convert it from a String to a URL object. 

The second part, where we change it from `.implicit` to `.general` is related to the callback URI changes made in 0.7. In 0.7, we made a documentation change in #186 to only document the `.general` callback URI, since all three flows are coming back from the end server. We also provided more specific instructions to developers on how to set the callback URI. This here solidifies the change by removing `.implicit` as the callback URI string used in these tests.

I think I should probably update the configuration's `CallbackURI` API so that it doesn't have the `type` parameter anymore.

Reply:
Ah, OK. My changes here are a bit confusing. I don't think I should have made this change, because it really doesn't matter. 

As long as we standardize callback URIs (which it looks like we will), we don't need this configuration option anymore. This would probably live in the `Authenticator` itself. 

I am going to add the backwards-compatibility changes @tyvsmith suggested in #203 for `Configuration`, and suggest that we merge this in the meantime.